concent_api/core/payments/report.py

```diff
@@ -0,0 +1,759 @@
+                            f'differs from the settlement payment with tx_hash {settlement_payment.tx_hash} '
+                            f'closure time value {settlement_payment.closure_time}.',
+                )
+            )
+
+
+def check_error_more_than_one_subtask_result_of_the_same_type_for_subtask(
+    errors: List[Error],
+    matched_objects: List,
+    additional_verification_payment: CoverAdditionalVerificationEvent,
+) -> None:
+    """ Add error if there is more than two SubtaskResultsSettled or SubtaskResultsRejected for the same subtask. """
+    if (
+        any(
+            [subtask_result.response_type == PendingResponse.ResponseType.SubtaskResultsSettled.name  # pylint: disable=no-member
+             for subtask_result in matched_objects]
+        ) and
+        any(
+            [subtask_result.response_type == PendingResponse.ResponseType.SubtaskResultsRejected.name  # pylint: disable=no-member
+             for subtask_result in matched_objects]
+        )
+    ):
+        errors.append(
+            Error(
+                matching_error=MatchingError.MORE_THAN_ONE_SUBTASK_RESULT_FOR_SUBTASK,
+                message=f'Subtask related with CoverAdditionalVerificationEvent with '
+                        f'tx_hash {additional_verification_payment.tx_hash} '
+                        f'and subtask_id {additional_verification_payment.subtask_id} '
+                        f'has both SubtaskResultsSettled or SubtaskResultsRejected.',
+            )
+        )
+
+
+def check_error_deposit_claim_exists_for_settlement_payment(errors: List[Error], tx_hash: str) -> None:
+    """ Add error if there is DepositClaim related with settlement payment. """
+    if DepositClaim.objects.filter(tx_hash=tx_hash).exists():
+        errors.append(
+            Error(
+                matching_error=MatchingError.DEPOSIT_CLAIM_RELATED_TO_SETTLEMENT_PAYMENT_EXISTS,
+                message=f'DepositClaim still exists for transaction with tx_hash {tx_hash}',
+            )
+        )
+
+
+def check_error_deposit_claim_exists_for_forced_payment(errors: List[Error], tx_hash: str) -> None:
+    """ Add error if there is DepositClaim related with forced payment. """
+    if DepositClaim.objects.filter(tx_hash=tx_hash).exists():
+        errors.append(
+            Error(
+                matching_error=MatchingError.DEPOSIT_CLAIM_RELATED_TO_FORCED_PAYMENT_EXISTS,
+                message=f'DepositClaim still exists for transaction with tx_hash {tx_hash}',
+            )
+        )
+
+
+def check_warning_deposit_claim_exists_for_missing_match(
+    warnings: List[Warnings],
+    subtask_id: str,
+    concent_use_case: ConcentUseCase,
+) -> None:
+    """
+    Add warning if message cannot be matched with a payment
+    but there's a DepositClaim that indicates that the transaction is pending.
+    """
+    if DepositClaim.objects.filter(
+        subtask_id=subtask_id,
+        concent_use_case=concent_use_case,
+    ).exists():
```

Comment:
If `tx_hash` is `None`, this could be an error because in that situation it's certain that something failed - Concent tried to make a payment and even claimed the funds but did not finalize it.